etcdserver/api/membership/cluster.go

```diff
@@ -279,12 +289,23 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 				plog.Panicf("unmarshal member should never fail: %v", err)
 			}
 		}
-		for _, u := range m.PeerURLs {
-			if urls[u] {
-				return ErrPeerURLexists
+		// A ConfChangeAddNode to a existing learner node promotes it to a voting member.
+		if members[id] != nil && members[id].IsLearner && !m.IsLearner {
+			if members[id] == nil {
```

Comment:
Why not just `return nil` ?
The out condition is `members[id] != nil` and the inner condition is  `members[id] == nil`

Reply:
uhh, my mistake